dcos/dcos-ui
# [spike] DCOS-51050: Services: Reappearing error before the configuration is submitted

Hide cosmos errors when editing the form.

Closes https://jira.mesosphere.com/browse/DCOS-51050

## Testing
1. Go to Catalog tab.
2. Select a package and enter some error in the configuration that Cosmos would catch, for example in chronos, go to "Chronos" tab and enter "/Chronos" as ID (invalid because of capital letter).
3. Go to "Review & Run" and try to submit.
4. An error should appear. As you go back, the error should disappear. As you click "Edit Config", it should disappear. It shouldn't appear even when going back to the review screen before clicking the submit button.
5. Fix the error and submit. The error shouldn't appear for a second.

## Trade-offs
Also reworked the way this removal works.

## Dependencies
None.

## Screenshots
### Before
![Peek 2019-03-29 14-21](https://user-images.githubusercontent.com/40791275/55232990-cdba7400-522f-11e9-91fc-3fe7b14a960e.gif)

### After
![Peek 2019-03-29 14-28](https://user-images.githubusercontent.com/40791275/55233011-ddd25380-522f-11e9-837d-6248aecd427d.gif)
